src/mscorlib/shared/System/Threading/AsyncLocal.cs

```diff
@@ -124,14 +124,29 @@ internal static class AsyncLocalValueMap
     {
         public static IAsyncLocalValueMap Empty { get; } = new EmptyAsyncLocalValueMap();
 
+        public static bool IsEmpty(IAsyncLocalValueMap asyncLocalValueMap)
+        {
+            Debug.Assert(asyncLocalValueMap != null);
```

Comment:
Maybe also assert that if asyncLocalValueMap != Empty then it's also not of type EmptyAsyncLocalValueMap.  We want to avoid the possibility of a bug where we end up with a different instance than the singleton.